Python/ast_unparse.c

```diff
@@ -67,6 +70,21 @@ append_repr(_PyUnicodeWriter *writer, PyObject *obj)
     if (!repr) {
```

Comment:
I suggest to add float code path *before* the PyObject_Repr() call.

```
    if (PyFloat_CheckExact(obj) && Py_IS_INFINITY(PyFloat_AS_DOUBLE(obj))) {
        return _PyUnicodeWriter_WriteStr(writer, _str_inf); 
    }
    PyObject *repr = PyObject_Repr(obj);
    ...
```

While we're modifying append_repr(), would you mind "int ret" definition to where it's used, at the end?

```
    int ret = _PyUnicodeWriter_WriteStr(writer, repr); 
```